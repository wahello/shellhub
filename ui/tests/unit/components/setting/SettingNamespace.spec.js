import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import { ValidationProvider, ValidationObserver } from 'vee-validate';
import flushPromises from 'flush-promises';
import Vuetify from 'vuetify';
import SettingNamespace from '@/components/setting/SettingNamespace';
import '@/vee-validate';

describe('SettingNamespace', () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  localVue.use(Vuex);
  localVue.component('ValidationProvider', ValidationProvider);
  localVue.component('ValidationObserver', ValidationObserver);

  document.body.setAttribute('data-app', true);

  let wrapper;

  const idOwner = '6';
  const idNotOwner = '10';
  const owner = true;
  const hasTenant = true;
  const isEnterpriseOwner = true;
  const isEnterprise = true;
  const textRole = ['Owner', 'Member', 'Member'];

  const countDevicesHasNamespacePercent = {
    maxDevices: 0,
    percent: 0,
  };

  const namespace = {
    name: 'namespace3',
    members: [{ id: '6', name: 'user6' }, { id: '7', name: 'user7' }, { id: '8', name: 'user8' }],
    owner: '6',
    tenant_id: 'e359bf484715',
    devices_count: 1,
    max_devices: 3,
  };

  const openNamespace = {
    name: 'namespace3',
    members: [{ id: '6', name: 'user6' }, { id: '7', name: 'user7' }, { id: '8', name: 'user8' }],
    owner: '6',
    tenant_id: 'e359bf484715',
    devices_count: 1,
    max_devices: -1,
  };

  const invalidNamespaces = [
    '\'', '"', '!', '@', '#', '$', '%', '¨', '&', '*', '(', ')', '-', '_', '=', '+', '´', '`', '[',
    '{', '~', '^', ']', ',', '<', '..', '>', ';', ':', '/', '?',
  ];

  const invalidMinAndMaxCharacters = [
    's', 'sh', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
  ];

  const storeNotOwner = new Vuex.Store({
    namespaced: true,
    state: {
      namespace,
      id: idNotOwner,
      owner: !owner,
      tenant: '',
    },
    getters: {
      'namespaces/get': (state) => state.namespace,
      'auth/id': (state) => state.id,
      'namespaces/owner': (state) => state.owner,
      'auth/tenant': (state) => state.tenant,
    },
    actions: {
      'namespaces/put': () => {},
      'namespaces/get': () => {},
      'namespaces/removeUser': () => {},
      'snackbar/showSnackbarSuccessAction': () => {},
      'snackbar/showSnackbarErrorAction': () => {},
      'security/get': () => {},
    },
  });

  const storeOwner = new Vuex.Store({
    namespaced: true,
    state: {
      namespace,
      id: idOwner,
      owner,
      tenant: openNamespace.tenant_id,
    },
    getters: {
      'namespaces/get': (state) => state.namespace,
      'auth/id': (state) => state.id,
      'namespaces/owner': (state) => state.owner,
      'auth/tenant': (state) => state.tenant,
    },
    actions: {
      'namespaces/put': () => {},
      'namespaces/get': () => {},
      'namespaces/removeUser': () => {},
      'snackbar/showSnackbarSuccessAction': () => {},
      'snackbar/showSnackbarErrorAction': () => {},
      'security/get': () => {},
    },
  });

  const storeOwnerOpen = new Vuex.Store({
    namespaced: true,
    state: {
      namespace: openNamespace,
      id: idOwner,
      owner,
      tenant: openNamespace.tenant_id,
    },
    getters: {
      'namespaces/get': (state) => state.namespace,
      'auth/id': (state) => state.id,
      'namespaces/owner': (state) => state.owner,
      'auth/tenant': (state) => state.tenant,
    },
    actions: {
      'namespaces/put': () => {},
      'namespaces/get': () => {},
      'namespaces/removeUser': () => {},
      'snackbar/showSnackbarSuccessAction': () => {},
      'snackbar/showSnackbarErrorAction': () => {},
      'security/get': () => {},
    },
  });

  ///////
  // In this case, hosted version tests
  ///////

  describe('Hosted version', () => {
    beforeEach(() => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValue('e359bf484715');

      wrapper = mount(SettingNamespace, {
        store: storeOwner,
        localVue,
        stubs: ['fragment'],
        mocks: {
          $env: {
            isEnterprise: true,
          },
          $stripe: {
            elements: () => ({
              create: () => ({
                mount: () => null,
              }),
            }),
          },
        },
        vuetify,
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Compare data with default value', () => {
      expect(wrapper.vm.name).toEqual(namespace.name);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.isOwner).toEqual(owner);
      expect(wrapper.vm.owner).toEqual(namespace.owner);
      expect(wrapper.vm.namespace).toEqual(namespace);
      expect(wrapper.vm.tenant).toEqual(namespace.tenant_id);
      expect(wrapper.vm.isEnterpriseOwner).toEqual(isEnterpriseOwner);
      expect(wrapper.vm.isEnterprise).toEqual(isEnterprise);
    });
    it('Process data in methods', () => {
      let percent = 0;
      if (namespace.max_devices >= 0) {
        percent = (wrapper.vm.countDevicesHasNamespace() / namespace.max_devices) * 100;
      }
      countDevicesHasNamespacePercent.maxDevices = namespace.max_devices;
      countDevicesHasNamespacePercent.percent = percent;

      expect(wrapper.vm.hasTenant()).toEqual(hasTenant);
      expect(wrapper.vm.countDevicesHasNamespace()).toEqual(namespace.devices_count);
      expect(wrapper.vm.countDevicesHasNamespacePercent())
        .toEqual(countDevicesHasNamespacePercent);
    });

    //////
    // HTML validation
    //////

    //////
    // In this case, invalid RFC1123.
    //////

    invalidNamespaces.forEach((inamespace) => {
      it(`Shows invalid namespace error for ${inamespace}`, async () => {
        wrapper.setData({ name: inamespace });
        await flushPromises();

        const validator = wrapper.vm.$refs.providerName;

        await validator.validate();
        expect(validator.errors[0]).toBe('You entered an invalid RFC1123 name');
      });
    });

    //////
    // In this case, password should be 3-30 characters long.
    //////

    invalidMinAndMaxCharacters.forEach((character) => {
      it(`Shows invalid namespace error for ${character}`, async () => {
        wrapper.setData({ name: character });
        await flushPromises();

        const validator = wrapper.vm.$refs.providerName;

        await validator.validate();
        expect(validator.errors[0]).toBe('Your namespace should be 3-30 characters long');
      });
    });

    it('Show validation messages', async () => {
      //////
      // In this case, validate fields required.
      //////

      wrapper.setData({ name: '' });
      await flushPromises();

      let validator = wrapper.vm.$refs.providerName;

      await validator.validate();
      expect(validator.errors[0]).toBe('This field is required');

      //////
      // In this case, must not contain dots.
      //////

      wrapper.setData({ name: 'ShelHub.' });
      await flushPromises();

      validator = wrapper.vm.$refs.providerName;

      await validator.validate();
      expect(validator.errors[0]).toBe('The name must not contain dots');
    });

    it('Renders the template with components', () => {
      expect(wrapper.find('[data-test="settingOwnerInfo-component"]').exists()).toBe(false);
    });
    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="tenant-span"]').text()).toEqual(namespace.tenant_id);

      //////
      // Check rendering of member names in the list.
      //////

      namespace.members.forEach((member) => {
        expect(wrapper.find(`[data-test="${member.name}-list"]`).text()).toEqual(member.name);
      });

      //////
      // Check owner fields rendering.
      //////

      expect(wrapper.find('[data-test="owner-p"]').text()).toEqual('Owner');

      expect(wrapper.find('[data-test="editOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="userOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="deleteOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="securityOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="notTheOwner-span"]').exists()).toEqual(false);
      expect(wrapper.findAll('[data-test="removeMember-btn"]').length).toEqual(namespace.members.length - 1);
      expect(wrapper.find('[data-test="role-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="newMember-div"]').exists()).toEqual(true);
    });
  });

  ///////
  // In this case, not the owner of this namespace
  ///////

  describe('Not the owner of this namespace', () => {
    beforeEach(() => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValue('e359bf484715');

      wrapper = mount(SettingNamespace, {
        localVue,
        store: storeNotOwner,
        stubs: ['fragment'],
        mocks: {
          $env: {
            isEnterprise: true,
          },
          $stripe: {
            elements: () => ({
              create: () => ({
                mount: () => null,
              }),
            }),
          },
        },
        vuetify,
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Compare data with default value', () => {
      expect(wrapper.vm.name).toEqual('');
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.isOwner).toEqual(!owner);
      expect(wrapper.vm.owner).toEqual(namespace.owner);
      expect(wrapper.vm.namespace).toEqual(namespace);
      expect(wrapper.vm.tenant).toEqual('');
      expect(wrapper.vm.isEnterpriseOwner).toEqual(!isEnterpriseOwner);
      expect(wrapper.vm.isEnterprise).toEqual(isEnterprise);
    });
    it('Process data in methods', () => {
      let percent = 0;
      if (namespace.max_devices >= 0) {
        percent = (wrapper.vm.countDevicesHasNamespace() / namespace.max_devices) * 100;
      }
      countDevicesHasNamespacePercent.maxDevices = namespace.max_devices;
      countDevicesHasNamespacePercent.percent = percent;

      expect(wrapper.vm.hasTenant()).toEqual(!hasTenant);
      expect(wrapper.vm.countDevicesHasNamespace()).toEqual(namespace.devices_count);
      expect(wrapper.vm.countDevicesHasNamespacePercent())
        .toEqual(countDevicesHasNamespacePercent);
    });

    //////
    // HTML validation
    //////

    it('Renders the template with components', () => {
      expect(wrapper.find('[data-test="settingOwnerInfo-component"]').exists()).toBe(true);
    });
    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="tenant-span"]').text()).toEqual('');

      //////
      // Check owner fields rendering.
      //////

      expect(wrapper.find('[data-test="owner-p"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="editOperation-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="userOperation-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="deleteOperation-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="securityOperation-div"]').exists()).toEqual(false);
      expect(wrapper.findAll('[data-test="removeMember-btn"]').length).toEqual(0);
      expect(wrapper.find('[data-test="role-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="newMember-div"]').exists()).toEqual(false);
    });
  });

  ///////
  // In this case, open version tests
  ///////

  describe('Open version', () => {
    beforeEach(() => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValue('e359bf484715');

      wrapper = mount(SettingNamespace, {
        localVue,
        store: storeOwnerOpen,
        stubs: ['fragment'],
        mocks: {
          $env: {
            isEnterprise: false,
          },
          $stripe: {
            elements: () => ({
              create: () => ({
                mount: () => null,
              }),
            }),
          },
        },
        vuetify,
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Compare data with default value', () => {
      expect(wrapper.vm.name).toEqual(namespace.name);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.isOwner).toEqual(owner);
      expect(wrapper.vm.owner).toEqual(namespace.owner);
      expect(wrapper.vm.namespace).toEqual({ ...namespace, max_devices: -1 });
      expect(wrapper.vm.tenant).toEqual(namespace.tenant_id);
      expect(wrapper.vm.isEnterpriseOwner).toEqual(!isEnterpriseOwner);
      expect(wrapper.vm.isEnterprise).toEqual(!isEnterprise);
    });
    it('Process data in methods', () => {
      let percent = 0;
      if (namespace.max_devices >= 0) {
        percent = (wrapper.vm.countDevicesHasNamespace() / namespace.max_devices) * 100;
      }
      countDevicesHasNamespacePercent.maxDevices = namespace.max_devices;
      countDevicesHasNamespacePercent.percent = percent;

      expect(wrapper.vm.hasTenant()).toEqual(hasTenant);
      expect(wrapper.vm.countDevicesHasNamespace()).toEqual(namespace.devices_count);
      expect(wrapper.vm.countDevicesHasNamespacePercent())
        .toEqual({ ...countDevicesHasNamespacePercent, maxDevices: -1, percent: 0 });
    });

    //////
    // HTML validation
    //////

    it('Renders the template with components', () => {
      expect(wrapper.find('[data-test="settingOwnerInfo-component"]').exists()).toBe(false);
    });
    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="tenant-span"]').text()).toEqual(namespace.tenant_id);

      //////
      // Check rendering of member names in the list.
      //////

      namespace.members.forEach((member) => {
        expect(wrapper.find(`[data-test="${member.name}-list"]`).text()).toEqual(member.name);
      });

      //////
      // Check owner fields rendering.
      //////

      expect(wrapper.find('[data-test="owner-p"]').exists()).toEqual(false);

      expect(wrapper.find('[data-test="editOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="userOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="deleteOperation-div"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test="securityOperation-div"]').exists()).toEqual(false);
      expect(wrapper.find('[data-test="notTheOwner-span"]').exists()).toEqual(false);
      expect(wrapper.findAll('[data-test="removeMember-btn"]').exists()).toEqual(false);
      expect(wrapper.findAll('[data-test=role-div]').wrappers.reduce((ac, v) => {
        ac.push(v.text());
        return ac;
      }, [])).toEqual(textRole);
      expect(wrapper.find('[data-test="newMember-div"]').exists()).toEqual(false);
    });
  });
});
