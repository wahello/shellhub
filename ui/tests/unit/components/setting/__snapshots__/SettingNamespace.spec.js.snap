// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingNamespace Hosted version Renders the component 1`] = `
<fragment-stub>
  <div class="container">
    <div class="row mt-4 align-center justify-center">
      <div class="col-sm-8 col">
        <!---->
        <div class="mt-6">
          <div class="row mb-2">
            <div class="col-md-auto col">
              <div class="v-card v-sheet theme--light elevation-0 rounded-0">
                Tenant ID:
              </div>
            </div>
            <div class="ml-auto col-md-auto col">
              <div class="auto v-card v-sheet theme--light elevation-0 rounded-0"><span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><span data-test="tenant-span">
                    e359bf484715
                  </span> <i aria-hidden="true" class="v-icon notranslate v-icon--right mdi mdi-content-copy theme--light" data-clipboard-click-handler="$3" data-clipboard-success-handler="$4"></i></span></span></div>
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
        </div>
        <div data-test="editOperation-div" class="mt-6"><span><div class="row"><div class="col"><h3>
                  Namespace
                </h3></div> <div class="spacer"></div> <div class="ml-auto col-md-auto col"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default"><span class="v-btn__content">
                  Rename Namespace
                </span></button></div>
      </div>
      <div class="mt-4 mb-2"><span><div class="v-input ml-3 v-input--is-label-active v-input--is-dirty theme--light v-text-field"><div class="v-input__control"><div class="v-input__slot"><div class="v-text-field__slot"><label for="input-70" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Name</label><input required="required" data-test="name-text" id="input-70" type="text"></div></div><div class="v-text-field__details"><div class="v-messages theme--light"><transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub></div></div></div></div></span></div></span>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    </div>
    <div data-test="userOperation-div" class="mt-6">
      <div class="row">
        <div class="col">
          <h3>
            Members
          </h3>
        </div>
        <div class="spacer"></div>
        <div data-test="newMember-div">
          <div class="ml-auto col-md-auto col">
            <fragment-stub><button type="button" class="ml-12 v-btn v-btn--outlined theme--light v-size--default" data-test="add-btn"><span class="v-btn__content">
    Add Member
  </span></button>
              <div role="dialog" class="v-dialog__container">
                <!---->
              </div>
            </fragment-stub>
          </div>
        </div>
      </div>
      <div class="mt-5">
        <div role="list" class="v-list v-sheet theme--light">
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user6-list" class="v-list-item__title">
                  user6
                </div>
              </div>
              <div class="spacer"></div>
              <div>
                <div class="ml-auto col-md-auto col">
                  <p data-test="owner-p" class="mr-3">
                    Owner
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user7-list" class="v-list-item__title">
                  user7
                </div>
              </div>
              <div class="spacer"></div>
              <div>
                <div class="ml-auto col-md-auto col"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default" data-test="removeMember-btn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">delete</i><span class="v-tooltip v-tooltip--bottom"><!----></span></span></button></div>
              </div>
            </div>
          </div>
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user8-list" class="v-list-item__title">
                  user8
                </div>
              </div>
              <div class="spacer"></div>
              <div>
                <div class="ml-auto col-md-auto col"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default" data-test="removeMember-btn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">delete</i><span class="v-tooltip v-tooltip--bottom"><!----></span></span></button></div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    </div>
    <div data-test="securityOperation-div" class="mt-6">
      <form novalidate="novalidate" class="v-form">
        <div class="row">
          <div class="mb-6 col">
            <h3 class="mb-5">
              Security
            </h3>
            <div class="ml-3">
              <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-96" role="checkbox" type="checkbox" value="">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-96" class="v-label theme--light" style="left: 0px; position: relative;">Enable session record</label>
                  </div>
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
              <p>
                Session record is a feature that allows you to check logged activity when
                connecting to a device.
              </p>
            </div>
          </div>
        </div>
      </form>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    </div>
    <div data-test="deleteOperation-div" class="mt-6">
      <h3 class="mb-5">
        Danger Zone
      </h3>
      <div class="row mt-4 mb-2">
        <div class="ml-3 col">
          <h4>
            Delete this namespace
          </h4>
          <div class="ml-2">
            <p>
              After deleting a namespace, there is no going back. Be sure.
            </p>
          </div>
        </div>
        <div class="ml-auto mb-4 col-md-auto col">
          <fragment-stub><button type="button" class="v-btn v-btn--outlined theme--light v-size--default red--text text--darken-1" data-test="delete-btn"><span class="v-btn__content">
    Delete namespace
  </span></button>
            <div role="dialog" class="v-dialog__container">
              <!---->
            </div>
          </fragment-stub>
        </div>
      </div>
    </div>
  </div>
  </div>
  </div>
</fragment-stub>
`;

exports[`SettingNamespace Not the owner of this namespace Renders the component 1`] = `
<fragment-stub>
  <div class="container">
    <div class="row mt-4 align-center justify-center">
      <div class="col-sm-8 col">
        <div style="text-align: center;" data-test="settingOwnerInfo-component">
          <div data-test="message-div">
            <h3 class="pl-6"><span>
        You're not the owner of this namespace.
      </span>
              <p data-test="contactUser-p">
                Contact user6 user for more information.
              </p>
            </h3>
          </div> <br>
        </div>
        <div class="mt-6">
          <div class="row mb-2">
            <div class="col-md-auto col">
              <div class="v-card v-sheet theme--light elevation-0 rounded-0">
                Tenant ID:
              </div>
            </div>
            <div class="ml-auto col-md-auto col">
              <div class="auto v-card v-sheet theme--light elevation-0 rounded-0"><span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><span data-test="tenant-span">
                    
                  </span> <i aria-hidden="true" class="v-icon notranslate v-icon--right mdi mdi-content-copy theme--light" data-clipboard-click-handler="$89" data-clipboard-success-handler="$90"></i></span></span></div>
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
        </div>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
  </div>
</fragment-stub>
`;

exports[`SettingNamespace Open version Renders the component 1`] = `
<fragment-stub>
  <div class="container">
    <div class="row mt-4 align-center justify-center">
      <div class="col-sm-8 col">
        <!---->
        <div class="mt-6">
          <div class="row mb-2">
            <div class="col-md-auto col">
              <div class="v-card v-sheet theme--light elevation-0 rounded-0">
                Tenant ID:
              </div>
            </div>
            <div class="ml-auto col-md-auto col">
              <div class="auto v-card v-sheet theme--light elevation-0 rounded-0"><span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><span data-test="tenant-span">
                    e359bf484715
                  </span> <i aria-hidden="true" class="v-icon notranslate v-icon--right mdi mdi-content-copy theme--light" data-clipboard-click-handler="$103" data-clipboard-success-handler="$104"></i></span></span></div>
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
        </div>
        <div data-test="editOperation-div" class="mt-6"><span><div class="row"><div class="col"><h3>
                  Namespace
                </h3></div> <div class="spacer"></div> <div class="ml-auto col-md-auto col"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default"><span class="v-btn__content">
                  Rename Namespace
                </span></button></div>
      </div>
      <div class="mt-4 mb-2"><span><div class="v-input ml-3 v-input--is-label-active v-input--is-dirty theme--light v-text-field"><div class="v-input__control"><div class="v-input__slot"><div class="v-text-field__slot"><label for="input-2229" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Name</label><input required="required" data-test="name-text" id="input-2229" type="text"></div></div><div class="v-text-field__details"><div class="v-messages theme--light"><transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub></div></div></div></div></span></div></span>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    </div>
    <div data-test="userOperation-div" class="mt-6">
      <div class="row">
        <div class="col">
          <h3>
            Members
          </h3>
        </div>
        <div class="spacer"></div>
        <!---->
      </div>
      <div class="mt-5">
        <div role="list" class="v-list v-sheet theme--light">
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user6-list" class="v-list-item__title">
                  user6
                </div>
              </div>
              <div class="spacer"></div>
              <div data-test="role-div">
                <div class="ml-auto col-md-auto col">
                  <p data-test="role-text">
                    Owner
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user7-list" class="v-list-item__title">
                  user7
                </div>
              </div>
              <div class="spacer"></div>
              <div data-test="role-div">
                <div class="ml-auto col-md-auto col">
                  <p data-test="role-text">
                    Member
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div tabindex="-1" role="listitem" class="v-list-item theme--light">
            <div class="row">
              <div class="ml-auto col-md-auto col"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
              <div class="col">
                <div data-test="user8-list" class="v-list-item__title">
                  user8
                </div>
              </div>
              <div class="spacer"></div>
              <div data-test="role-div">
                <div class="ml-auto col-md-auto col">
                  <p data-test="role-text">
                    Member
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    </div>
    <!---->
    <div data-test="deleteOperation-div" class="mt-6">
      <h3 class="mb-5">
        Danger Zone
      </h3>
      <div class="row mt-4 mb-2">
        <div class="ml-3 col">
          <h4>
            Delete this namespace
          </h4>
          <div class="ml-2">
            <p>
              After deleting a namespace, there is no going back. Be sure.
            </p>
          </div>
        </div>
        <div class="ml-auto mb-4 col-md-auto col">
          <fragment-stub><button type="button" class="v-btn v-btn--outlined theme--light v-size--default red--text text--darken-1" data-test="delete-btn"><span class="v-btn__content">
    Delete namespace
  </span></button>
            <div role="dialog" class="v-dialog__container">
              <!---->
            </div>
          </fragment-stub>
        </div>
      </div>
    </div>
  </div>
  </div>
  </div>
</fragment-stub>
`;
